core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/FileSinkTestRun.java

```diff
@@ -214,7 +215,7 @@ private void runPipeline(FileFormat format, @Nullable String delimiter, FileCons
     }
   }
 
-  private void validateDatasetSchema(FileFormat format) {
+  private void validateDatasetSchema(FileFormat format) throws IOException {
     // if a schema was provided for the sink verify that the external dataset has the given schema
     Map<String, String> metadataProperties =
       metadataAdmin.getProperties(MetadataScope.SYSTEM,
```

Comment:
what surprises me is that hydrator-plugins depends directly on MetadataAdmin. That is an internal class of CDAP. So this can break anytime internal CDAP classes change?

Reply:
The MetadataAdmin (in addition to other Admin classes) are exposed by our TestBase:
https://github.com/cdapio/cdap/blob/develop/cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java#L911

But yes, MetadataAdmin is a class in cdap-app-fabric module which is exposed via the TestBase, and users' unit tests could break compilation when this internal class changes.

Reply:
I see. I guess it's a necessary for now, but we should fix that in the future. Please merge this to unblock ITN,